src/plugins/discover/public/context_awareness/hooks/use_profile_accessor.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useCallback } from 'react';
+import { getMergedAccessor } from '../composable_profile';
+import type { GetProfilesOptions } from '../profiles_manager';
+import { useProfiles } from './use_profiles';
+import type { Profile } from '../types';
+
+export const useProfileAccessor = <TKey extends keyof Profile>(
```

Comment:
Same comment as before. 

> I think it might be worth adding a couple of lines of JSDoc as it is not 100% clear when this utilities do exactly from the code.